server/src/test/java/org/elasticsearch/persistent/PersistentTasksExecutorIT.java

```diff
@@ -173,6 +180,42 @@ public void testPersistentActionWithNoAvailableNode() throws Exception {
         assertEquals(removeFuture.get().getId(), taskId);
     }
 
+    public void testPersistentActionWithNonClusterStateCondition() throws Exception {
+        PersistentTasksClusterService persistentTasksClusterService =
+            internalCluster().getInstance(PersistentTasksClusterService.class, internalCluster().getMasterName());
+        // Speed up rechecks to a rate that is quicker than what settings would allow
+        persistentTasksClusterService.setRecheckInterval(TimeValue.timeValueMillis(10));
```

Comment:
This will still make a slow test. How about calling the setTimeInterval method in PersistentTasksClusterService directly and settings it to something much faster?

Reply:
I changed it to 1ms here and the test runs in 85ms on my laptop now.